packages/abstract-core/src/objects/fee.rs

```diff
@@ -71,6 +71,76 @@ impl Fee {
     }
 }
 
+/// A wrapper around Coin to help handle fixed fees (with multiples).
+#[cosmwasm_schema::cw_serde]
+pub struct FixedFee {
+    /// Fee to be paid for a unit of a service
+    fee: Coin,
+}
+
+impl FixedFee {
+    pub fn new(fee: Coin) -> Self {
+        FixedFee { fee }
+    }
+    /// Allows to collect the fee multiple times
+    /// E.g., for namespaces, you want to charge the number of claimed namespaces times the fee for 1 namespace
+    pub fn quantity(&self, qty: u128) -> Self {
+        Self {
+            fee: Coin {
+                denom: self.fee.denom.clone(),
+                amount: self.fee.amount * Uint128::from(qty),
+            },
+        }
+    }
+
+    //
+    pub fn charge(&self, msg_info: &MessageInfo) -> AbstractResult<()> {
+        if self.fee.amount.is_zero() {
+            return Ok(());
+        }
+        if msg_info.funds.len() != 1
+            || msg_info.funds[0].denom != self.fee.denom
+            || self.fee.amount != msg_info.funds[0].amount
+        {
+            return Err(AbstractError::Fee(format!(
+                "Invalid fee payment sent. Expected {}, sent {:?}",
+                self.fee, msg_info.funds
+            )));
+        }
+        Ok(())
+    }
+
+    pub fn charge_mut(&self, msg_info: &mut MessageInfo) -> AbstractResult<()> {
```

Comment:
Would be nice if these functions returned the amount that they charged, which can then be used to preform transfers if the dev wants to. Also add doc comments.